src/cmakeProject.ts

```diff
@@ -1989,7 +1989,37 @@ export class CMakeProject {
      * Implementation of `cmake.buildType`
      */
     async currentBuildType(): Promise<string | null> {
-        return this.variantManager.activeVariantOptions.buildType || null;
+        let buildType: string | null = null ;
+        if (this.useCMakePresets) {
+            if (this.buildPreset) {
+                if (this.buildPreset.configuration) {
+                    // The `configuration` is set for multi-config generators, and is optional for single-config generators.
+                    // Get the first value for multi-config generators
+                    buildType = this.buildPreset.configuration.split(';')[0];
```

Comment:
According to the Presets spec, `this.buildPreset.configuration` should not be an array that needs to be split.  It should just be the string which has the `CMAKE_BUILD_TYPE` you are looking for.

Reply:
Not fixed yet